src/python/pants/engine/process.py

```diff
@@ -477,7 +477,6 @@ class BashBinary(BinaryPath):
 
 @dataclass(frozen=True)
 class BashBinaryRequest:
-    rationale: str
```

Comment:
Since the rationale provided little value in this case, the change probably makes sense. That said, if we expected more fluent use of dataclasses / more thought about the types used in rule signatures / Gets, this could just have been, IIUC:
```python
@dataclass(frozen=True)
class BashBinaryRequest:
    rationale: str = dataclasses.field(compare=False)
```